include/rocksdb/memory_allocator.h

```diff
@@ -74,4 +74,21 @@ extern Status NewJemallocNodumpAllocator(
     JemallocAllocatorOptions& options,
     std::shared_ptr<MemoryAllocator>* memory_allocator);
 
+// PersistentMemoryAllocator is an interface that a client can implement to
+// supply custom persistent memory allocation, deallocation, initialization and
+// finlization functions. All methods should be thread-safe.
+class PersistentMemoryAllocator : public MemoryAllocator {
```

Comment:
Should `PersistentMemoryAllocator` inherit `void* Allocate(...)`, `Deallocate(void*)` methods from `MemoryAllocator`? I am dubious about using `void*` to represent persistent pointers. It is true that a 64-bit integer may be sufficient to represent a large address space, but some other PMEM libraries use more than that. There is no consensus on this yet.